auditlog/models.py

```diff
@@ -208,6 +215,58 @@ def _get_pk_value(self, instance):
             pk = self._get_pk_value(pk)
         return pk
 
+    def _get_serialized_data_or_none(self, instance):
+        from .registry import auditlog
+
+        opts = auditlog.get_serialize_options(instance.__class__)
+        if not opts["serialize_data"]:
+            return None
+
+        model_fields = auditlog.get_model_fields(instance.__class__)
+        kwargs = opts.get("serialize_kwargs", {})
+
+        if opts["serialize_auditlog_fields_only"]:
+            kwargs.setdefault(
+                "fields", self._get_applicable_model_fields(instance, model_fields)
+            )
+
+        data = dict(json.loads(serializers.serialize("json", (instance,), **kwargs))[0])
+
+        if model_fields["mask_fields"]:
+            data = self._mask_serialized_fields(data, model_fields)
+
+        return data
+
+    def _get_applicable_model_fields(
+        self, instance, model_fields: Dict[str, List[str]]
+    ) -> List[str]:
+        include_fields = model_fields["include_fields"]
+        exclude_fields = model_fields["exclude_fields"]
+        all_field_names = [field.name for field in instance._meta.fields]
+
+        if not include_fields and not exclude_fields:
+            return all_field_names
+
+        return list(set(include_fields or all_field_names).difference(exclude_fields))
+
+    def _mask_serialized_fields(
```

Comment:
What if model A has a foreign key to model B, and model B is configured with `mask_fields`? :)

Can we add a note somewhere in the docs that `mask_fields` doesn't affect fields embedded in serialized related objects? A user may assume that and we should preempt this assumption.

Reply:
I have a few thoughts here.

1. Forward relationships would simply show the FK unless the developer takes specific actions by setting up natural keys. Using confidential information within natural keys would show up in a whole lot of places and would therefore be unadvisable beyond the scope of the auditlog.
2. Serializing reverse relationships into nested lists are not supported in this PR and therefore `mask_fields` from those related tables will never show up in the serialized data.
3. Most importantly, I had added support for nested masked values from related tables in [this commit](https://github.com/jazzband/django-auditlog/pull/412/commits/23541cf7fef8a7d50c1534f4d8741d11ed691936) but was asked to remove it due to unnecessary complexity (which I tend to agree is the correct decision).

Reply:
> Forward relationships would simply show the FK

Ah, I misunderstood your other comment and thought that the forward relationship would have had a complete (or incomplete :upside_down_face:) representation of the related object. Please ignore this thread then. :+1: